src/terraformpy/resource_collections.py

```diff
@@ -90,6 +91,28 @@ def finalize_resources(self):
         """
         pass
 
+    # We need to override this here, since we changed the way the instantiation
+    # works for models. Sadly, the underlying implementaion had to be
+    # duplicated entirely
+    @classmethod
+    def get_mock_object(cls, context=None, overrides=None):
```

Comment:
To avoid duplicating this all here we could add a new condition in our init method that checks:

```
if len(kwargs) == 0 and len(args) == 1 and isinstance(args[0], dict):
    kwargs = args.pop()
```

We already have a bunch of specific logic to handle schematics-isms there and I feel like it is the lesser evil than duplicating this method.

Reply:
Done, though I wasn't able to verify every spot that could be potentially affected, but the tests are green.